### PR TITLE
synchrona: add support for synchrona/rpi eeprom

### PR DESCRIPTION
**Pull request by ranechita**
For AD-SYNCHRONA14-EBZ the serial number and other relevant information had to stored on the RPI4 EEPROM. This PR aims to add support for writing and reading fru-files in Raspberry Pi EEPROM (512KB bin files) without altering the RPI bootloader, by operating only on the last ~256 bytes

### CODE REVIEW

**Comment by rgetz** [comment]
Is there a way to make this more generic? Can you describe what the EEPROM contents look like on your Pi?

**Comment by rgetz** [comment]
And what I mean by that - is add a command line argument (-O --offset <bytes>) so this isn't specific to that board, or that EEPROM, and then nothing is hard coded in the actual fru.c or fru.h (it all mostly ends up in the operating system section)...

Thanks
-Robin

**Comment by ranechita** [comment]
The Pi EEPROM file contains the rpi bootloader binary in the first half, and the rest of the file contains just FFs which I figured we could use to store our fru-file contents. The changes are already mostly done in the operating system section, except for 2 defines in fru.h. 
Adding a command line argument for offset seems like a very good option to make everything more generic, but how would we differentiate between the input file and output file? For instance, if we want to write a master file (-i 256 bytes) into the rpi eeprom file (-o 512KB), or the other way around, is there a way we could manage setting the offset for the output file but not for the input?

**Comment by ranechita** [comment]
v2 - After some considerations, I added this version in which there is a separate command line argument for the input file offset (-I <offset_in>) and for the output file offset (-O <offset_out>)

File: operating-system.c [hunk 1]
```diff
@@ -153,9 +159,16 @@ void write_FRU(struct FRU_DATA *fru, char * file_name, bool packed)
 		if (tmp != len)
 			printf_err ("Didn't write entire file\n");
 	} else {
-		if((fp = fopen(file_name, "wb")) == NULL)
+		if((fp = fopen(file_name, "r+b")) == NULL)
 			printf_err("Cannot open file.\n");
 
+		fseek(fp, 0L, SEEK_END);
+		long size = ftell(fp);
+		rewind(fp);
```
Comment (nunojsa) [inline operating-system.c]: Consider using `fstat()`. While this works on posix based systems, this is not "best practices" (and portable way) to get the file size (even more when 'b' is involved).

**Comment by rgetz** [comment]
I never merged things - can you squash these together?

**Comment by ranechita** [comment]
v3 : 

- squashed commits
- updated commit description

**Review by AlexandraTrifan — APPROVED** [review]

**Pull request closed without merging**